app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagItemViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.app.devoptions.featureflags
+
+import androidx.annotation.ColorInt
+import androidx.databinding.ObservableField
+import org.oppia.android.app.model.FeatureFlagId
+import org.oppia.android.app.model.SyncStatus
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.models.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.locale.OppiaLocale
+import javax.inject.Inject
+
+/** [ViewModel] for displaying a feature flag for the recycler view in [FeatureFlagsFragment]. */
+class FeatureFlagItemViewModel @Inject constructor(
+  val featureFlagId: FeatureFlagId,
+  val currentValue: Boolean,
+  val syncStatus: SyncStatus,
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val resourceHandler: AppLanguageResourceHandler
+
+) : ObservableViewModel() {
+
+  val isChecked: ObservableField<Boolean> = ObservableField(currentValue)
+  val featureFlagDisplayName: ObservableField<String> =
+    ObservableField(getFeatureFlagDisplayName(featureFlagId))
+  val syncStatusDisplayText: ObservableField<String> =
+    ObservableField(getSyncStatusText())
+
+  var onToggleCallback: ((FeatureFlagId, Boolean) -> Unit)? = null
+
+  fun onUserToggle() {
+    isChecked.set(!isChecked.get()!!)
+    onToggleCallback?.invoke(featureFlagId, isChecked.get()!!)
+  }
+
+  @ColorInt
+  val backgroundColor: Int = retrieveBackgroundColor().toInt()
+  private fun getFeatureFlagDisplayName(
+    id: FeatureFlagId
+  ): String {
+    return machineLocale.run {
+      when (id) {
+        FeatureFlagId.UNRECOGNIZED,
+        FeatureFlagId.FEATURE_FLAG_ID_UNSPECIFIED -> "Unknown Feature"
+
+        else ->
```

Comment:
Let's skip the line breaks in the `when` statements to align with the existing codebase style.
```suggestion
        FeatureFlagId.FEATURE_FLAG_ID_UNSPECIFIED -> "Unknown Feature"
        else ->
```

Reply:
updated